src/pages/Posts.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import './Post.css';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+import { toast } from 'react-hot-toast'
+
+const Posts = () => {
+    const [posts, setPosts] = useState([]);
+    const [comments, setComments] = useState([]);
+    const [newComment, setNewComment] = useState('');
+    const handleCommentSubmit = async (id) => {
+        setComments((prevComments) => [...prevComments, newComment]);
+        console.log(newComment, id)
+        try {
+            const response = await axios.post(`/posts/:${id}/comment`, { comment: newComment });
+            console.log(response.data)
+        } catch (error) {
+            console.error('Error fetching:', error.message);
+        }
+        setNewComment('');
+    };
+    const fetchPosts = async () => {
+        try {
+            const response = await axios.get(`/posts`);
+            setPosts(response.data);
+        } catch (error) {
+            console.error('Error fetching:', error.message);
+        }
+    };
+    const handleDelete = async (id) => {
+        try {
+            const response = await axios.delete(`/posts/${id}`);
+            toast.success(response.data.message)
+            window.location.reload()
+        } catch (error) {
+            toast.error('Error fetching sightings:', error);
+        }
+    };
+    const handleLike = async (id) => {
+        try {
+            const response = await axios.put(`/posts/${id}/like`);
+            toast.success(response.data.message)
+        } catch (error) {
+            toast.error('Error fetching sightings:', error);
+        }
+    };
+    useEffect(() => {
+        fetchPosts();
+    }, []);
+
+
+    return (
+        <>
+            <Link to="/addpost">
+                <button >Add Post
+
+                </button>
+            </Link>
+            {posts && posts?.map((p, i) => (
+
+                <div
+                    key={i}
+                    className="post">
+                    <h2 className="post-title">{p?.title}</h2>
+                    <p className="post-desc">{p?.desc}</p>
+                    {/* <p className="post-created-at">Created at: {createdAt}</p> */}
+                    <div className="post-actions">
+                        <button onClick={() => handleLike(p?.id)}>Like ({p?.likes})</button>
+                        <button className='del' onClick={() => handleDelete(p?.id)}>Delete </button>
+                    </div>
+                    <div className="post-comments">
+                        {/* {console.log(p?.comment)} */}
+                        <h3>{p?.comment}</h3>
```

Comment:
why the optional chaining here? Usually we would use that on nested properties to avoid errors on accessing properties on undefined. `p` should be always defined here as it is a single post. If that would be undefined, then your `posts` array would be a bit odd. Example `const posts = [undefined,Post{...}, Post{...}, undefined]`. If your array looks like that, optional chaining would be okay to do I guess, but then your choice of how to structure the array is a bit questionable